hazelcast-client/src/main/java/com/hazelcast/client/nearcache/ClientHeapNearCache.java

```diff
@@ -136,6 +138,9 @@ public void run() {
                         } finally {
                             canEvict.set(true);
                         }
+                        if (cache.size() >= maxSize && canEvict.compareAndSet(true, false)) {
+                            executionService.execute(this);
```

Comment:
I am sending another pr. I will set canEvict to true in case of RejectedExecutionException.